### ISSUE TITLE
Is it possible to have more than 1 item displayed on carousel view ?

### ISSUE THREAD

**Issue by Kelve17** [AndreiMisiukevich/CardView#206]
I want to display 3 items at every swipe the user makes? Is it possible ?

**Comment by AndreiMisiukevich** [comment]
Hi, yes, sure. Check readme — coverflowview.

Or just make model which will contain 3 submodels. And create ItemTemplate with 3 views

**Comment by Kelve17** [comment]
I tried coverflowview but it does not show 3 items on same row. Some item is shown cutted, i mean not full. and on android when swiped to left, the last item disapears.

**Comment by AndreiMisiukevich** [comment]
Okay, use another approach.

**Comment by Kelve17** [comment]
Is it possible to make the way u said ? with 1 model having 3 submodels ? Because i want to swipe on submodels.

**Comment by AndreiMisiukevich** [comment]
@Kelve17 yes, of course. That's trivial task

**Comment by Kelve17** [comment]
Sry, im newbie. Would you take a look at my code and see if im in the right way ?

**Comment by AndreiMisiukevich** [comment]
@Kelve17 yes, i can

**Comment by Kelve17** [comment]
Alright. Thanks for the patience!
` public class Example : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler PropertyChanged;


        private string _titulo;
        public string Titulo
        {
            get
            {
                return _titulo;
            }
            set
            {
                _titulo = value;
                OnPropertyChanged(nameof(Titulo));
            }
        }

        private string _titulo2;
        public string Titulo2
        {
            get
            {
                return _titulo2;
            }
            set
            {
                _titulo2 = value;
                OnPropertyChanged(nameof(Titulo2));
            }
        }

        private string _titulo3;
        public string Titulo3
        {
            get
            {
                return _titulo3;
            }
            set
            {
                _titulo3 = value;
                OnPropertyChanged(nameof(Titulo3));
            }
        }

        private Color _cor;
        public Color Cor
        {
            get
            {
                return _cor;
            }
            set
            {
                _cor = value;
                OnPropertyChanged(nameof(Cor));
            }
        }


        private Color _cor2;
        public Color Cor2
        {
            get
            {
                return _cor2;
            }
            set
            {
                _cor2 = value;
                OnPropertyChanged(nameof(Cor2));
            }
        }



        private Color _cor3;
        public Color Cor3
        {
            get
            {
                return _cor3;
            }
            set
            {
                _cor3 = value;
                OnPropertyChanged(nameof(Cor3));
            }
        }


        public Example(string a,string c, string d, Color b, Color e, Color f)
        {
            Titulo = a;
            Titulo2 = c;
            Titulo3 = d;
            Cor = b;
            Cor2 = e;
            Cor3 = f;
        }

        #region INotifyPropertyChanged Implementation
        void OnPropertyChanged([CallerMemberName] string propertyName = "")
        {
            if (PropertyChanged == null)
                return;

            PropertyChanged.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }
        #endregion
    }

    public class TesteViewModel : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler PropertyChanged;

        private ObservableCollection<Example> _fonte;
        public ObservableCollection<Example> Fonte
        {
            get
            {
                return _fonte;
            }
            set
            {
                _fonte = value;
                OnPropertyChanged(nameof(Fonte));
            }
        }

        public TesteViewModel()
        {
            Fonte = new ObservableCollection<Example>()
            {
                new Example("Gratidão","Vitórias","Objectivos do ano", Color.Red, Color.Green, Color.Blue),
                new Example("Vitórias","Objectivos do ano","Gratidão", Color.Green, Color.Blue, Color.Red),
                new Example("Objectivos do ano","Gratidão","Vitórias", Color.Blue, Color.Red, Color.Green),
            };
        }

        #region INotifyPropertyChanged Implementation
        void OnPropertyChanged([CallerMemberName] string propertyName = "")
        {
            if (PropertyChanged == null)
                return;

            PropertyChanged.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }
        #endregion

    }

    [XamlCompilation(XamlCompilationOptions.Compile)]
    public partial class teste : ContentPage
    {

        public teste()
        {
            BindingContext = new TesteViewModel();
            InitializeComponent();
        }


    }`

` <StackLayout>
            
            <card:CoverFlowView PositionShiftPercentage="0.5" PositionShiftValue="1" x:Name="carrouusel" IsCyclical="True" AreAnimationsEnabled="True" ItemsSource="{Binding Fonte}">
                <card:CoverFlowView.ItemTemplate>
                    <DataTemplate>
                        <ContentView Padding="20">
                            <StackLayout Orientation="Horizontal" Spacing="10">
                                <Frame HeightRequest="50" WidthRequest="100" Padding="0" HasShadow="True" IsClippedToBounds="true" 
                                   CornerRadius="0" BackgroundColor="{Binding Cor}" VerticalOptions="Center" HorizontalOptions="Fill">
                                    <Label Text="{Binding Titulo}" TextColor="Black" HorizontalOptions="CenterAndExpand" VerticalOptions="CenterAndExpand"/>
                                </Frame>
                                <Frame HeightRequest="50" WidthRequest="100" Padding="0" HasShadow="True" IsClippedToBounds="true" 
                                   CornerRadius="0" BackgroundColor="{Binding Cor2}" VerticalOptions="Center" HorizontalOptions="Fill">
                                    <Label Text="{Binding Titulo2}" TextColor="Black" HorizontalOptions="CenterAndExpand" VerticalOptions="CenterAndExpand"/>
                                </Frame>
                                <Frame HeightRequest="50" WidthRequest="100" Padding="0" HasShadow="True" IsClippedToBounds="True" 
                                   CornerRadius="0" BackgroundColor="{Binding Cor3}" VerticalOptions="Center" HorizontalOptions="Fill">
                                    <Label Text="{Binding Titulo3}" TextColor="Black" HorizontalOptions="CenterAndExpand" VerticalOptions="CenterAndExpand"/>
                                </Frame>
                            </StackLayout>
                        </ContentView>
                    </DataTemplate>
                </card:CoverFlowView.ItemTemplate>
            </card:CoverFlowView>
            
        </StackLayout>`

**Comment by AndreiMisiukevich** [comment]
oh... okay. i will provide you pseudocode of what you really need. Then you will be play with it

ViewModel
```csharp
public class ViewModel {
	public List<Item> Items { get; } = new List<ItemsBunch>
	{
		new ItemsBunch
		{
			Item1 = new Item
			{
				Title = "1",
				Color = Color.Red
			},
			Item2 = new Item
			{
				Title = "2",
				Color = Color.Green
			},
			Item3 = new Item
			{
				Title = "3",
				Color = Color.Yellow
			}
		}
	}
}

public class ItemsBunch
{
	public Item1 { get; set; }
	public Item2 { get; set; }
	public Item3 { get; set; }
}

public class Item
{
	public string Title { get; set; }
	public Color Color { get; set; }
}
```

**Comment by AndreiMisiukevich** [comment]
and xaml

So, carouselView is what you need

```xaml
<card:CarouselView ItemsSource="{Binding Fonte}">
    <card:CarouselView.ItemTemplate>
        <DataTemplate>
            <ContentView>
                <StackLayout Orientation="Horizontal">

                    <Frame Padding="0" IsClippedToBounds="true" 
                       CornerRadius="0" BackgroundColor="{Binding Item1.Color}" VerticalOptions="Center" HorizontalOptions="FillAndExpand">
                        <Label Text="{Binding Item1.Title}" TextColor="Black" HorizontalOptions="CenterAndExpand" VerticalOptions="CenterAndExpand"/>
                    </Frame>
                    
                    <Frame Padding="0" IsClippedToBounds="true" 
                       CornerRadius="0" BackgroundColor="{Binding Item2.Color}" VerticalOptions="Center" HorizontalOptions="FillAndExpand">
                        <Label Text="{Binding Item2.Title}" TextColor="Black" HorizontalOptions="CenterAndExpand" VerticalOptions="CenterAndExpand"/>
                    </Frame>
                    
                    <Frame Padding="0" IsClippedToBounds="true" 
                       CornerRadius="0" BackgroundColor="{Binding Item3.Color}" VerticalOptions="Center" HorizontalOptions="FillAndExpand">
                        <Label Text="{Binding Item3.Title}" TextColor="Black" HorizontalOptions="CenterAndExpand" VerticalOptions="CenterAndExpand"/>
                    </Frame>

                </StackLayout>
            </ContentView>
        </DataTemplate>
    </card:CarouselView.ItemTemplate>
</card:CarouselView>
```

**Comment by Kelve17** [comment]
Hi @AndreiMisiukevich. With this aprouch, when i swipe to select the next `Item,` it will target the next `ItemBunch` and not the next `Item`.

**Comment by AndreiMisiukevich** [comment]
@Kelve17 okay, in this case check code in my samples - Coverflow sample. Also you can download source code and run the sample. 

https://github.com/AndreiMisiukevich/CardView/tree/master/PanCardViewSample

https://github.com/AndreiMisiukevich/CardView/blob/master/PanCardViewSample/PanCardViewSample/Views/CoverFlowSampleXamlView.xaml

**Comment by Kelve17** [comment]
I think the best option for me is to create a custom renderer based on your coverflow and display the 3 items per swipe. The only problem is that i have no idea on how to do such. Could u give me some tips or recommend some link where i could learn how to do such ?

**Comment by AndreiMisiukevich** [comment]
@Kelve17 show me please what do you try to achieve? Video / imagery

**Comment by Kelve17** [comment]
@AndreiMisiukevich ![2](https://user-images.githubusercontent.com/39514753/58918710-528ba780-8723-11e9-9926-b16725db7b65.jpg)
What i want to achieve is: In the top of this example mockup , there are 3 frames with label/image inside of ite. The 1st frame is the selected one. And the 2nd frame is the one that should be selected when the user swipes left. and the 3rd should be move to left 1 position until it eventually becomes the selected one.
Basically is a carousel view, but the next 2 pages that can be selected are always displayed fully (entire frame).

**Comment by Kelve17** [comment]
here : https://forums.xamarin.com/discussion/140962/xamarin-form-how-to-show-2-or-3-images-together-using-carousel u can have another example. But for me, i need the next 2 items to be always visible.

**Comment by AndreiMisiukevich** [comment]
@Kelve17 probably it can be achieved with coverflowView. but it's not so easy

**Comment by Kelve17** [comment]
somehow i managed to come very close to what i need with this :

```
            <card:CoverFlowView  PositionShiftPercentage="0.47" PositionShiftValue="80" x:Name="carrouusel" IsCyclical="True" AreAnimationsEnabled="True" ItemsSource="{Binding Fonte}">
                <card:CoverFlowView.ItemTemplate>
                    <DataTemplate>
                        <ContentView Padding="20">
                            <StackLayout Orientation="Horizontal" Spacing="10">
                                <Frame HeightRequest="50" WidthRequest="100" Padding="0" HasShadow="True" IsClippedToBounds="true" 
                                   CornerRadius="0" BackgroundColor="{Binding Cor}" VerticalOptions="Center" HorizontalOptions="Fill">
                                    <Label Text="{Binding Titulo}" TextColor="Black" HorizontalOptions="CenterAndExpand" VerticalOptions="CenterAndExpand"/>
                                </Frame>
                            </StackLayout>
                        </ContentView>
                    </DataTemplate>
                </card:CoverFlowView.ItemTemplate>
            </card:CoverFlowView>
```

But everytime i swipe to left, the last item disapears. It might be a bug i guess... Works fine when i swipe to right side.

**Comment by AndreiMisiukevich** [comment]
You always can try to debug ut
